src/main.rs

```diff
@@ -93,46 +102,299 @@ macro_rules! create_socket {
+
+        // TODO: Configure the ethernet controller
+        // Enable the ethernet peripheral.
+        //clocks.apb4.enr().modify(|_, w| w.syscfgen().set_bit());
+        //clocks.ahb1.enr().modify(|_, w| {
+        //    w.eth1macen().set_bit()
+        //        .eth1txen().set_bit()
+        //        .eth1rxen().set_bit()
+        //});
+
+        //dp.SYSCFG.pmcr.modify(|_, w| unsafe { w.epis().bits(0b100) }); // RMII
+
+        cp.SCB.enable_icache();
 
         init_log();
         // info!("Version {} {}", build_info::PKG_VERSION, build_info::GIT_VERSION.unwrap());
         // info!("Built on {}", build_info::BUILT_TIME_UTC);
         // info!("{} {}", build_info::RUSTC_VERSION, build_info::TARGET);
 
-        // c.schedule.tick(Instant::now()).unwrap();
+        let mut debug_pin = gpioc.pc6.into_push_pull_output();
+        debug_pin.set_low().unwrap();
+
+        let mut dac_pin = gpiob.pb15.into_push_pull_output();
+        dac_pin.set_low().unwrap();
+
+        // Configure timer 2 to trigger conversions for the ADC
+        let mut timer2 = dp.TIM2.timer(500.khz(), &mut clocks);
+        timer2.listen(hal::timer::Event::TimeOut);
 
-        let dp = c.device;
         init::LateResources {
-            spi: (dp.SPI1, dp.SPI2, dp.SPI4, dp.SPI5),
-            i2c: dp.I2C2,
-            ethernet_periph: (
-                dp.ETHERNET_MAC,
-                dp.ETHERNET_DMA,
-                dp.ETHERNET_MTL,
-            ),
+            adc1: adc1_spi,
+            dac1: dac1_spi,
+            adc2: adc2_spi,
+            dac2: dac2_spi,
+
+            dbg_pin: debug_pin,
+            dac_pin: dac_pin,
+            timer: timer2,
+
+            _eeprom_i2c: i2c2,
+//            ethernet_periph: (
+//                dp.ETHERNET_MAC,
+//                dp.ETHERNET_DMA,
+//                dp.ETHERNET_MTL,
+//            ),
+        }
+    }
+
+    #[task(binds = TIM2, resources = [dbg_pin, timer, adc1, adc2])]
+    fn tim2(mut c: tim2::Context) {
+        c.resources.timer.clear_uif_bit();
+        c.resources.dbg_pin.set_high().unwrap();
+
+        // Start a SPI transaction on ADC0 and ADC1
+        c.resources.adc1.lock(|adc| adc.spi.cr1.modify(|_, w| w.cstart().set_bit()));
+        c.resources.adc2.lock(|adc| adc.spi.cr1.modify(|_, w| w.cstart().set_bit()));
+
+        c.resources.dbg_pin.set_low().unwrap();
+    }
+
+    #[task(binds = SPI2, resources = [adc1, dac1, adc2, dac2, iir_state, iir_ch, dac_pin], priority = 2)]
+    fn adc_spi(c: adc_spi::Context) {
+        #[cfg(feature = "bkpt")]
+        cortex_m::asm::bkpt();
+
+        c.resources.dac_pin.set_high().unwrap();
+
+        let output_ch1 = {
+            let a: u16 = c.resources.adc1.read().unwrap();
+            let x0 = f32::from(a as i16);
+            let y0 = c.resources.iir_ch[0].update(&mut c.resources.iir_state[0], x0);
+            y0 as i16 as u16 ^ 0x8000
+        };
+        c.resources.adc1.spi.ifcr.write(|w| w.eotc().set_bit());
```

Comment:
In the old firmware I did this very early (before the read+processing+write) because of the SPI silicon erratum.

Reply:
This has been moved to the start of the transaction (although I do not believe the errata is cause for concern).